catalog/spatial/registry/registry-report-action-provider/src/main/java/org/codice/ddf/registry/report/action/provider/RegistryReportActionProvider.java

```diff
@@ -0,0 +1,226 @@
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.action.Action;
+import ddf.action.ActionProvider;
+import ddf.action.impl.ActionImpl;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.source.Source;
+
+public class RegistryReportActionProvider implements ActionProvider {
+
+    private String title;
+
+    private String description;
+
+    private static final String UNKNOWN_TARGET = "0.0.0.0";
+
+    private static final String REGISTRY_PATH = "/registries";
+
+    private static final String REPORT_PATH = "/report";
+
+    private static final String FORMAT = ".html";
+
+    private static final String SOURCE_ID_QUERY_PARAM = "?sourceId=";
+
+    private final String actionProviderId;
+
+    private ConfigurationAdmin configurationAdmin;
+
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(RegistryReportActionProvider.class);
+
+    public RegistryReportActionProvider(String id) {
+        this.actionProviderId = id;
+    }
+
+    public <T> List<Action> getActions(T subject) {
+
+        if (canHandle(subject)) {
+            if (subject instanceof Metacard) {
+                return processSubject((Metacard) subject);
+            } else if (subject instanceof Source) {
+                return processSubject((Source) subject);
+            } else if (subject instanceof Configuration) {
+                return processSubject((Configuration) subject);
+            }
+        }
+        return Collections.emptyList();
+    }
+
+    private List<Action> processSubject(Metacard metacard) {
+
+        if (metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID) == null
+                || StringUtils.isBlank(metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                .getValue()
+                .toString())) {
+            LOGGER.info("No registry id given. No action to provide.");
+            return Collections.emptyList();
+        }
+
+        try {
+            String registryId =
+                    URLEncoder.encode(metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                            .getValue()
+                            .toString(), (StandardCharsets.UTF_8).toString());
+            String sourceId = URLEncoder.encode(getSource(metacard),
+                    (StandardCharsets.UTF_8).toString());
+            Action action = getAction(registryId, sourceId);
+
+            if (action == null) {
+                return Collections.emptyList();
+            }
+
+            return Collections.singletonList(action);
+
+        } catch (UnsupportedEncodingException e) {
+            LOGGER.info("Unsupported Encoding exception", e);
+            return Collections.emptyList();
+        }
+    }
+
+    private List<Action> processSubject(Source source) {
+        String configId = source.getId();
```

Comment:
It looks like the only thing being used from the `Source` is his id. Maybe this should just take the id directly also/instead? Looking at, say, `CatalogFramework` it looks like we try to hide the `Source`s themselves. @pklinef can you elaborate on how this API would be used?